File: GameServer/Content/Data/LeagueSandbox-Default/Champions/Ezreal/W.lua
Vector2 = require 'Vector2' -- include 2d vector lib 

function onFinishCasting()
    local current = Vector2:new(owner.X, owner.Y)
    local to = (Vector2:new(spell.X, spell.Y) - current):normalize()
    local range = to * 1000
    local trueCoords = current + range
	
    addProjectile("EzrealEssenceFluxMissile", trueCoords.x, trueCoords.y)
end

function applyEffects()
    if owner.Team ~= getTarget().Team then
        dealMagicalDamage(25+spellLevel*45+owner:GetStats().AbilityPower.Total*0.8)
	    addParticleTarget(owner, "Ezreal_essenceflux_tar.troy", getTarget())
    end
end

File: Content/LeagueSandbox-Scripts/Champions/Annie/W.cs
using GameServerCore.Enums;
using GameServerCore.Domain.GameObjects;
using static LeagueSandbox.GameServer.API.ApiFunctionManager;
using GameServerCore.Domain;
using LeagueSandbox.GameServer.Scripting.CSharp;
using System.Numerics;

namespace Spells
{
    public class Incinerate : IGameScript
    {
        public void OnActivate(IObjAiBase owner)
        {
        }

        public void OnDeactivate(IObjAiBase owner)
        {
        }

        public void OnStartCasting(IObjAiBase owner, ISpell spell, IAttackableUnit target)
        {
        }

        public void OnFinishCasting(IObjAiBase owner, ISpell spell, IAttackableUnit target)
        {
            var current = new Vector2(owner.X, owner.Y);
            var to = Vector2.Normalize(new Vector2(spell.X, spell.Y) - current);
            var range = to * 625;
            var trueCoords = current + range;

            spell.AddCone("Incinerate", trueCoords.X, trueCoords.Y, 24.76f);
            AddParticle(owner, "IIncinerate_buf.troy", trueCoords.X, trueCoords.Y);
            FaceDirection(owner, trueCoords, false);
            spell.SpellAnimation("SPELL2", owner);
            AddParticleTarget(owner, "Incinerate_cas.troy", owner);
        }

        public void ApplyEffects(IObjAiBase owner, IAttackableUnit target, ISpell spell, IProjectile projectile)
        {
            var ap = owner.Stats.AbilityPower.Total * 0.8f;
            var damage = 70 + spell.Level * 45 + ap;

            target.TakeDamage(owner, damage, DamageType.DAMAGE_TYPE_MAGICAL, DamageSource.DAMAGE_SOURCE_SPELL, false);

        }

        public void OnUpdate(double diff)
        {
        }
    }
}
